src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,58 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private final K [] keys;
+    private final V [] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getIndex(key);
+        if (index
+                != -1 || isFull()) {
```

Comment:
if the index is not equal to -1 that means your array contains value by index and it could be rewritten.
```suggestion
        if (index != -1) {
```